app/core/models.py

```diff
@@ -329,3 +332,39 @@ class Sdg(AbstractBaseModel):
 
     def __str__(self):
         return f"{self.name}"
+
+
+class Affiliation(AbstractBaseModel):
+    """
+    Sponsor/partner relationships stored in this table are project-dependent.
+    They cannot be both a sponsor and a partner for the same project,
+    so if is_sponsor is partner they are a project partner,
+    if is_sponsor is sponsor they are a project sponsor.
+    """
+
+    SPONSOR = "sponsor"
+    PARTNER = "partner"
+
+    AFFILIATION_CHOICES = [
+        (SPONSOR, "Sponsor"),
+        (PARTNER, "Partner"),
+    ]
+
+    affiliate = models.ForeignKey(SponsorPartner, on_delete=models.CASCADE)
+    project = models.ForeignKey(Project, on_delete=models.CASCADE)
+    ended_at = models.DateTimeField("Ended at", null=True, blank=True)
+    is_sponsor = models.CharField(max_length=20, choices=AFFILIATION_CHOICES)
```

Comment:
I think this is the type of affiliation. But `type` is a python builtin word so we have to choose something else like `kind` or another word.

Reply:
I was originally thinking of using python enums as choices. There's some advantages to using enums.